src/masternodeman.cpp

```diff
@@ -1689,6 +1691,47 @@ void CMasternodeMan::UpdatedBlockTip(const CBlockIndex *pindex)
     }
 }
 
+void CMasternodeMan::WarnMasternodeDaemonUpdates()
+{
+    static bool fWarned = false;
+
+    if (fWarned || !size() || !masternodeSync.IsMasternodeListSynced())
```

Comment:
shouldn't this be protected by `cs` as well?

Reply:
Not sure, should be safe imo. But better be safe than sorry, pushed the fix ;)

Reply:
I was mostly worried about the size() call as it accesses mapMasternodes